pact-tests/pact-tests/modref-recursion.repl

```diff
@@ -0,0 +1,32 @@
+
+(interface call
+
+  (defun callF:integer (m:module{call}))
+  )
+
+(module knot1 g
+  (defcap g () true)
+
+  (implements call)
+
+  (defun callF:integer (m:module{call})
+    (+ 1 2) ; call something so it costs gas
+    (m::callF knot1)
+  )
+  )
+
+(module knot2 g
+  (defcap g () true)
+
+  (implements call)
+
+  (defun callF:integer (m:module{call})
+    (+ 1 2) ; call something so it costs gas
+    (m::callF knot2)
+  )
+  )
+
+(env-gasmodel "table")
+(env-gaslimit 10) ; ensures test does not run forever in case recursion breaks
+
+(expect-failure "Recursion should fail @ runtime" (knot2.callF knot1))
```

Comment:
let's add a case for direct self-recursion as well. Also, is mutual recursion covered for both modrefs and two concrete mutually recursive calls?